sdk/samples/iot/iot_sample_common.h

```diff
@@ -0,0 +1,190 @@
+char iot_sample_x509_trust_pem_file_path_buffer[256];
+
+typedef struct
+{
+  az_span hub_hostname;
+  az_span provisioning_id_scope;
+  az_span hub_device_id;
+  az_span provisioning_registration_id;
+  az_span hub_sas_key;
+  az_span provisioning_sas_key;
+  uint32_t sas_key_duration_minutes;
+  az_span x509_cert_pem_file_path;
+  az_span x509_trust_pem_file_path;
+} iot_sample_environment_variables;
+
+typedef enum
+{
+  PAHO_IOT_HUB,
+  PAHO_IOT_PROVISIONING
+} iot_sample_type;
+
+typedef enum
+{
+  PAHO_IOT_HUB_C2D_SAMPLE,
+  PAHO_IOT_HUB_METHODS_SAMPLE,
+  PAHO_IOT_HUB_PNP_COMPONENT_SAMPLE,
+  PAHO_IOT_HUB_PNP_SAMPLE,
+  PAHO_IOT_HUB_SAS_TELEMETRY_SAMPLE,
+  PAHO_IOT_HUB_TELEMETRY_SAMPLE,
+  PAHO_IOT_HUB_TWIN_SAMPLE,
+  PAHO_IOT_PROVISIONING_SAMPLE,
+  PAHO_IOT_PROVISIONING_SAS_SAMPLE
+} iot_sample_name;
+
+/*
+ * @brief Reads in environment variables set by user for purposes of running sample.
+ *
+ * @param[in] type The enumerated type of the sample.
+ * @param[in] name The enumerated name of the sample.
+ * @param[out] out_env_vars A pointer to the struct containing all read-in environment variables.
+ *
+ * @return An #az_result value indicating the result of the operation.
+ * @retval #AZ_OK All required environment variables successfuly read-in.
+ * @retval #AZ_ERROR_ARG Sample type or name is undefined, or environment variable is not set.
+ * @retval #AZ_RETURN_IF_NOT_ENOUGH_SIZE Not enough space set aside to store environment variable.
+ */
+az_result iot_sample_read_environment_variables(
+    iot_sample_type type,
+    iot_sample_name name,
+    iot_sample_environment_variables* out_env_vars);
+
+/*
+ * @brief Builds an MQTT endpoint c-string for an Azure IoT Hub or provisioning service.
+ *
+ * @param[in] type The enumerated type of the sample.
+ * @param[in] env_vars A pointer to environment variable struct.
+ * @param[out] out_endpoint A pointer to char buffer containing the built c-string.
+ * @param[in] endpoint_size The size of the char buffer to be filled.
+ *
+ * @return An #az_result value indicating the result of the operation.
+ * @retval #AZ_OK MQTT endpoint successfully created.
+ * @retval #AZ_ERROR_ARG Sample type is undefined.
+ * @retval #AZ_ERROR_INSUFFICIENT_SPAN_SIZE Buffer size is not large enough to hold c-string.
+ */
+az_result iot_sample_create_mqtt_endpoint(
+    iot_sample_type type,
+    iot_sample_environment_variables const* env_vars,
+    char* out_endpoint,
+    size_t endpoint_size);
+
+/*
+ * @brief Sleep for given seconds.
+ *
+ * @param[in] seconds Number of seconds to sleep.
+ */
+void iot_sample_sleep_for_seconds(uint32_t seconds);
+
+/*
+ * @brief Return total seconds passed including supplied minutes.
+ *
+ * @param[in] minutes Number of minutes to include in total seconds returned.
+ * @return Total time in seconds.
+ */
+uint32_t iot_sample_get_epoch_expiration_time_from_minutes(uint32_t minutes);
+
+/*
+ * @brief Generate the b64 encoded and signed signature using HMAC-SHA256 signing.

```

Comment:
Is b64 commonly known shorthand? I think we should be explicit. Same with parameter names.

```suggestion
 * @brief Generate the base64 encoded and signed signature using HMAC-SHA256 signing.

```

Reply:
in the context it would but yea expanding it out here is fine

Reply:
I am renaming function names and parameters to be more specific.  Can you please review my next commit? Thanks.